libcaf_core/caf/logger.hpp

```diff
@@ -265,84 +202,19 @@ class CAF_CORE_EXPORT logger : public ref_counted {
   /// registered.
   static logger* current_logger();
 
-private:
+protected:
   // -- constructors, destructors, and assignment operators --------------------
 
   logger(actor_system& sys);
 
-  // -- called by the actor_system when running with a test coordinator --------
-
-  void inline_output(bool value) noexcept {
-    cfg_.inline_output = value;
-  }
-
-  // -- initialization ---------------------------------------------------------
+  virtual void init(actor_system_config& cfg) = 0;
 
-  void init(actor_system_config& cfg);
+  virtual void inline_output(bool value) = 0;
 
-  bool open_file();
-
-  // -- event handling ---------------------------------------------------------
-
-  void handle_event(const event& x);
-
-  void handle_file_event(const event& x);
-
-  void handle_console_event(const event& x);
-
-  void log_first_line();
-
-  void log_last_line();
-
-  // -- thread management ------------------------------------------------------
-
-  void run();
-
-  void start();
-
-  void stop();
-
-  // -- member variables -------------------------------------------------------
-
-  // Configures verbosity and output generation.
-  config cfg_;
-
-  // Filters events by component name before enqueuing a log event. Intersection
-  // of file_filter_ and console_filter_ if both outputs are enabled.
-  std::vector<std::string> global_filter_;
-
-  // Filters events by component name for file output.
-  std::vector<std::string> file_filter_;
-
-  // Filters events by component name for console output.
-  std::vector<std::string> console_filter_;
+  virtual void start() = 0;
 
   // References the parent system.
   actor_system& system_;
```

Comment:
There shouldn't be any state in the interface. Also shouldn't inherit from `ref_counted` for that reason.

For the ref counting, there should be `ref_logger` and `deref_logger` as pure virtual functions. You can use https://github.com/actor-framework/actor-framework/blob/master/libcaf_core/caf/flow/coordinated.hpp as an example.